src/aiodynamo/expressions.py

```diff
@@ -0,0 +1,537 @@
+
+    def __invert__(self) -> Condition:
+        return NotCondition(self)
+
+    @abc.abstractmethod
+    def encode(self, params: Parameters) -> str:
+        pass
+
+
+@dataclass(frozen=True)
+class NotCondition(Condition):
+    base: Condition
+
+    def encode(self, params: Parameters) -> str:
+        return f"NOT {self.base.encode(params)}"
+
+
+@dataclass(frozen=True)
+class AndCondition(Condition):
+    lhs: Condition
+    rhs: Condition
+
+    def encode(self, params: Parameters) -> str:
+        return f"({self.lhs.encode(params)} AND {self.rhs.encode(params)})"
+
+
+@dataclass(frozen=True)
+class OrCondition(Condition):
+    lhs: Condition
+    rhs: Condition
+
+    def encode(self, params: Parameters) -> str:
+        return f"({self.lhs.encode(params)} OR {self.rhs.encode(params)})"
+
+
+@dataclass(frozen=True)
+class DoesNotExist(Condition):
+    field: F
+
+    def encode(self, params: Parameters) -> str:
+        return f"attribute_not_exists({params.encode_path(self.field.path)})"
+
+
+@dataclass(frozen=True)
+class Exists(Condition):
+    field: F
+
+    def encode(self, params: Parameters) -> str:
+        return f"attribute_exists({params.encode_path(self.field.path)})"
+
+
+@dataclass(frozen=True)
+class AttributeTypeCondition(Condition):
+    field: F
+    attribute_type: AttributeType
+
+    def encode(self, params: Parameters) -> str:
+        return f"attribute_type({params.encode_path(self.field.path)}, {self.attribute_type.value})"
+
+
+@dataclass(frozen=True)
+class BeginsWith(Condition):
+    field: F
+    substr: str
+
+    def encode(self, params: Parameters) -> str:
+        return f"begins_with({params.encode_path(self.field.path)}, {params.encode_value(self.substr)})"
+
+
+@dataclass(frozen=True)
+class Between(Condition):
+    field: F
+    low: Any
+    high: Any
+
+    def encode(self, params: Parameters) -> str:
+        return f"{params.encode_path(self.field.path)} BETWEEN {params.encode_value(self.low)} AND {params.encode_value(self.high)}"
+
+
+@dataclass(frozen=True)
+class Contains(Condition):
+    field: F
+    value: Union[str, bytes, int, float, decimal.Decimal]
+
+    def encode(self, params: Parameters) -> str:
+        return f"contains({params.encode_path(self.field.path)}, {params.encode_value(self.value)})"
+
+
+@dataclass(frozen=True)
+class In(Condition):
+    field: F
+    values: Sequence[Any]
+
+    def encode(self, params: Parameters) -> str:
+        encoded_values = [
+            encoded
+            for encoded in (params.encode_value(value) for value in self.values)
+            if encoded is not None
+        ]
+        if len(encoded_values) < 1:
+            raise ValueError("IN Condition requires at least one value")
+        if len(encoded_values) > 100:
+            raise ValueError("IN Condition may not contain more than 100 values")
+        values = ",".join(encoded_values)
+        return f"{params.encode_path(self.field.path)} IN ({values})"
```

Comment:
not sure about parentheses here too... things like `a & x in (vvv) & b`

Reply:
`x in(y)` is a single thing in dynamo so no parens around it needed.